VoxelSniperCore/src/main/java/com/github/kevindagame/brush/TreeSnipeBrush.java

```diff
@@ -33,18 +33,7 @@ public TreeSnipeBrush() {
     }
 
     private void single(final SnipeData v, IBlock targetBlock) {
-        Undo undo = new Undo();
-
-        IBlock blockBelow = targetBlock.getRelative(BlockFace.DOWN);
-        IBlockState currentState = blockBelow.getState();
-        undo.put(currentState);
-        blockBelow.setMaterial(VoxelMaterial.GRASS_BLOCK, false);
-
-        this.getWorld().generateTree(targetBlock.getLocation(), this.treeType, undo);
-
-        blockBelow.setBlockData(currentState.getBlockData(), false);
-
-        v.owner().storeUndo(undo);
+        this.positions.add(targetBlock.getLocation());
```

Comment:
More block will be changed, although I don't know a good way to get all affected blocks

Reply:
Yeah nothing we can really do about this

Reply:
I might have an idea how to fix this, might try tonight

Reply:
https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/82/commits/ef5ed94bda016b7b5ebb5ef4d4f1c4f4f34b7067